### ISSUE TITLE
[Question]: which dimension should we calculate the norm across?

### ISSUE THREAD

**Issue by JunnYu** [nbasyl/DoRA#1]
Hello, I have noticed that in the paper, the normalization (norm) is applied to the dimension of `out_feature`, but in the implementation of PEFT, it is applied to the dimension of `in_feature`. Which one is correct?

d is out_feature,  k is in_feature, r is rank
weight.shape = [d, k]
lora_A.shape = [r, k]
lora_B.shape = [d, r]
![image](https://github.com/nbasyl/DoRA/assets/50394665/a98dc0da-7056-44c5-9a35-a4f0ba22295f)
https://github.com/huggingface/peft/blob/6dca6d22922fcb1ead828b5b3c146911d7b693fb/src/peft/tuners/lora/layer.py#L172-L175

weight.norm(dim=0) -> [1, k]
weight.norm(dim=1) -> [d, 1]  [PEFT Implement]

**Comment by nbasyl** [comment]
Hi, the normalization is always applied on the out_feature dimension. We did not formulate W in this figure the same as nn.Linear.weight. The shape of W in this figure is of the shape (in_features, out_features) and the shape of nn.Linear.weight is of the shape (out_features,in_features) which might be causing your confusion.

**Comment by JunnYu** [comment]
@nbasyl Thank you, I understand now.
